wordpress-mobile/WordPress-iOS
# Stats Revamp: Details Screen Date Picker

This PR adds functionality for the DatePicker on the Details screen as part of Stats Revamp.
 * The Date Picker now works and makes an API call + refreshes screen for the selected date interval period
 * DatePicker does not show when the tap originated from out of scope cards (ie Tags and Categories) that were not included in Stats Revamp v2

Ref #18601 

To test:
1. First, build and run, navigate to Stats Insights for a site and ensure that there is no option to add a Views & Visitors card like in s/shot below
2. Click around insights, period stats and ensure there are no crashes
3. From the insights screen tap View more on This Year.  There should be no changes compared to current app store version of the app
4. Repeat the same process for Tags and Categories, Followers and Posting Activity

Now enable both of the new stats feature flags and build and run again (`statsNewAppearance, statsNewInsights`)

1. Add New Stats Card = Views & Visitors.  New Views & Visitors Card and Graph should be added to the screen
2. Tap "This week" on the upper right of the Views / Visitors card.  Screen will segue / navigate to the Details card like below
3. Tap the DatePicker at top to go back in time + data will update and tap DatePicker to go forward in time
4. Navigate back to main insights screen and tap View More on This Year.  DatePicker should not be visible on the Details Screen
5. Repeat for Tags and Categories, Followers and Posting Activity


## Regression Notes
1. Potential unintended areas of impact
Stats

2. What I did to test those areas of impact (or what existing automated tests I relied on)
I manually tested and added a unit test

3. What automated tests I added (or what prevented me from doing so)
I added a unit test

PR submission checklist:
